puppet/chassis.rb

```diff
@@ -177,7 +177,24 @@ def self.install_extension(extension)
 			repo = extension
 		end
 
-		folder = @@dir + '/extensions/' + extension.split('/').last.gsub(/.git$/, '').downcase
+		# We need to add a version to allow for a seamless upgrade to Bionic Beaver.
+		if config["version"] >= 3
+			# Check for existing extensions that have a hyphen and remove them.
+			old_extensions = Dir.glob(@@dir + '/extensions/*')
+			old_extensions.each { |extension|
+				if extension.include? "-"
+					# Delete the old extension if it has a hyphen in it.
+					FileUtils.remove_dir(extension)
+					new_extension = extension.gsub(/-/, '_')
+					puts "We've upgraded #{extension} to #{new_extension}"
```

Comment:
"renamed" might be better to be more explicit, plus maybe we should link out somewhere to explain why we're doing it? (e.g. link to this issue/PR)

Reply:
Good call. Done!